File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    answer = 'Yes'
    for word in goal:
        if cards1 != [] and cards1[0] == word:
            cards1.pop(0)
        elif cards2 != [] and cards2[0] == word:
            cards2.pop(0)
        else:
            return 'No'
    return answer

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.swift
import Foundation

func solution(_ cards1:[String], _ cards2:[String], _ goal:[String]) -> String {
    
    var head1 = 0
    var head2 = 0
    
    for g in goal {
        if head1 < cards1.count && cards1[head1] == g {
            head1 += 1
        } else if head2 < cards2.count && cards2[head2] == g {
            head2 += 1
        } else {
            return "No"
        }
    }
    
    return "Yes"
}